### ISSUE TITLE
Create a top-level `ScriptExt` for .proj files

### ISSUE THREAD

**Issue by sandreenko** [dotnet/runtime#46235]
Add a top-level define for `ScriptExt`:
```
<ScriptExt Condition="'$(OS)' == 'Windows_NT'">.cmd</ScriptExt>
<ScriptExt Condition="'$(OS)' != 'Windows_NT'">.sh</ScriptExt>
```
to https://github.com/dotnet/runtime/blob/master/Build.proj

See the discussion in https://github.com/dotnet/runtime/pull/46225#discussion_r545955453

It will clear files that are currently using code like:
https://github.com/dotnet/runtime/blob/f99d251269c4acf1ccd04963d1268ae6a1e3417e/src/mono/mono.proj#L16
or
https://github.com/dotnet/runtime/blob/f99d251269c4acf1ccd04963d1268ae6a1e3417e/src/coreclr/crossgen-corelib.proj#L18
or
https://github.com/dotnet/runtime/blob/f99d251269c4acf1ccd04963d1268ae6a1e3417e/src/coreclr/runtime.proj#L34
etc.

**Comment by MCCshreyas** [comment]
Hi @sandreenko, Can you please assign this to me, I will get this fixed and will open a PR for same! 😄

**Comment by sandreenko** [comment]
> Hi @sandreenko, Can you please assign this to me, I will get this fixed and will open a PR for same! 😄

Sure!

**Comment by MCCshreyas** [comment]
Hi @sandreenko I have opened PR for the issue. I have added `ScriptExt` for Build.proj. 
> It will clear files that are currently using code like:

Do we need to remove those tags from those other proj files as well? Or the root option that we added in `Build.proj` will override those? 

Please let me know if there are any more changes required. Thanks.

**Comment by sandreenko** [comment]
> Do we need to remove those tags from those other proj files as well? Or the root option that we added in Build.proj will override those?

The root option will be available for all projects in subfolder, so you can delete the definitions from the children and use the top one where it is appropriate, I would grep for ".cmd" in "*.proj" files to see where it can be used.

From the first look these files can be improved:
```
Name	Size	Matches	Path	Ext	Encoding	Date modified
crossgen-corelib.proj	6.24 KB	2	src\coreclr	proj	ANSI	12/18/2020 8:54:23 PM
runtime.proj	2.78 KB	1	src\coreclr	proj	ANSI	12/18/2020 9:12:43 AM
mono.proj	38.8 KB	4	src\mono	proj	ANSI	12/18/2020 9:12:43 AM
```
just replace code like this:
```
      <DotNetCli>$([MSBuild]::NormalizePath('$(RepoRoot)', 'dotnet.sh'))</DotNetCli>
      <DotNetCli Condition="'$(OS)' == 'Windows_NT'">$([MSBuild]::NormalizePath('$(RepoRoot)', 'dotnet.cmd'))</DotNetCli>
```
with 
```
<DotNetCli>$([MSBuild]::NormalizePath('$(RepoRoot)', 'dotnet.$(ScriptExt)'))</DotNetCli>
```

**Comment by sandreenko** [comment]
won't be fixed, see discussion in #46248
